src/docs/arch-cursor.dox

```diff
@@ -3,13 +3,393 @@
+```
+    session->open_cursor(session, "table:main(x1,x4,x3,k2,k1,x2,x4)", NULL, NULL, &cursor);
+
+```
+In this case, the plan is this string (with spacing added):
+```
+0v n 1v s n 1v n 0k s n 0k n 0v s n 1v s n
+```
+To break it down,   
+- \c "0v" means use sub-cursor 0 and prepare to read values.
+- \c "n" means get the next value, that is \c "x1".
+- \c "1v" means use sub-cursor 1 and prepare to read values.
+- \c "s" means skip a value, that is \c "x3".
+- \c "n" means get the next value, that is \c "x4".
+- \c ...
+
+Notice that this plan requires many switches of cursors and several \c "s" (skip) operations.
+Each skip involves enough decoding of the data item to determine its length so its data can
+be skipped over.
+
+With the default (complete) projection, getting values is fast.  Using the default plan, the needed
+columns are pulled out of each subordinate cursor one by one, being copied to the caller's arguments.
+With projections, the simple algorithm following the plan works well if the columns in the projection
+are grouped by column group and requested in order.  Without that discipline, as in this example,
+the performance will not be optimal.
+
+The implementation of plan creation and execution resides the schema.
+
+@section arch_cursor_dump Dump cursors
+
+Dump cursors are used in two rather different ways.  Regular dump cursors retrieve the raw keys and values
+and translate bytes either as raw characters or translating to hex.  This flavor of dump cursor is used
+by the `wt dump` or `wt dump -x` utility.  JSON dump cursors do more sophisticated translation, returning a string that is a JSON
+formatted record, with the name for each key and its data and the name for each value and its corresponding data.
+Data is translated to either integral, floating point, or string depending on the format of the column in the schema.
+This flavor of dump cursor is used by the `wt dump -j` command as part of creating a JSON dump of WiredTiger URI.
+
+There is a single \c WT_CURSOR_DUMP struct that is used to implement both flavors.
+The dump code checks for the \c WT_CURSTD_DUMP_JSON flag and as needed, calls into functions
+in `src/cursor/cur_json.c` .  The code in that file also implements several external functions that are used
+by the `wt` utility when loading JSON-dumped files.  In particular, the \c __wt_json_token function
+returns individual JSON tokens from an input string.
+
+The JSON code used by the dump cursor uses some storage that hangs off of `cursor->json_private` that
+is typed as \c WT_CURSOR_JSON.  When a JSON flavored cursor is created, the list of key column names
+and value column names is populated in \c WT_CURSOR_JSON.  These names, obtained from the
+configuration string that created the table or file, are useful to have in advance,
+as they are used once per row to help fill out the JSON output.  The functions that get
+rows know to iterate these names and unpack the corresponding column data,
```

Comment:
know to iterate -> know how to iterate

Reply:
I removed "know to", so it's just `The functions that get rows iterate these names...`. Seems clearer.